### PR TITLE
Fixes API change from 2015-02-18

### PR DESCRIPTION
**Pull request by toadkicker**
This add backward compatibility and handles the current api's swap of 'cards' to 'sources'.

### CODE REVIEW

**Comment by eddywashere** [comment]
thanks @toadkicker!

**Comment by benrlodge** [comment]
I'm not too familiar with the stripe setup, but I'm getting an error on this: `TypeError: Cannot read property 'data' of undefined`.  It doesn't look like the customer object in the first expression is returning a cards property for me.

The `customer.sources.data[0]` does the trick for me though in any case, so I just swapped the order of the two expressions.

**Comment by toadkicker** [comment]
@benrlodge Stripe changed their API to 'sources' but as far as I could tell the rest of the data structure is the same. If `customer.cards` is false it should move onto try to set it as `customer.sources`. What's the node version you're running against?

File: server/models/plugins/stripe-customer.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ module.exports = exports = function stripeCustomer (schema, options) {
         user.stripe.customerId = customer.id;
       }
 
-      var card = customer.cards.data[0];
+      var card = customer.cards.data[0] || customer.sources.data[0];
```
Comment (eddywashere) [inline server/models/plugins/stripe-customer.js]: I missed this. it should be `var cardSource = customer.cards || customer.sources;`, then `var card = cardSource.data[0]`

**Comment by benrlodge** [comment]
It's not `customer.cards` that it's failing on, it's the `data` property on `cards`, `customer.cards.data`. Because the`cards` property doesn't exist, checking for any property on it will return undefined, as a reference is unresolvable if it's base is undefined. If we want to keep the expression check in that order, something like this should do the trick - `var card = customer.cards ? customers.cards.data[0] : customer.sources.data[0];`

**Comment by benrlodge** [comment]
https://github.com/eddywashere/node-stripe-membership-saas/pull/16

**Pull request merged**